teuthology/lock/ops.py

```diff
@@ -175,16 +176,19 @@ def unlock_many(names, user):
         locked_by=user,
         names=names,
     )
-    response = requests.post(
-        uri,
-        data=json.dumps(data),
-        headers={'content-type': 'application/json'},
-    )
-    if response.ok:
-        log.debug("Unlocked: %s", ', '.join(names))
-    else:
-        log.error("Failed to unlock: %s", ', '.join(names))
-    return response.ok
+    with safe_while(
+            sleep=1, increment=0.5, action=f'unlock_many {names}') as proceed:
```

Comment:
unlocking itself isn't contended, it's hitting an issue currently due to the postgres connection being in a bad state from a previous conflict. I believe this should be avoided with the SERIALIZABLE change to paddles, since it will always commit or rollback, leaving the postgres connections in a good state for the next thread.

The retry here is more for safety in case of future unforeseen errors.

Reply:
I honestly look forward of deploying this SERIALIZABLE paddle change.

Reply:
Do we really need this change merged if the SERIALIZABLE resolves the issue?

Reply:
SERIALIZABLE fixes the issue of calls like unlock, which aren't contended, getting an error due to the postgres thread being in a bad state.

This PR is still necessary for handling contended cases, which from the paddles logs include update_inventory() and job status updates. For these contended cases, we'll continue to get errors from paddles, so we need to retry them from teuthology. I tried to get paddles to retry internally, but it turned out to be way more complex than this approach.